core/src/main/java/hudson/FilePath.java

```diff
@@ -415,7 +448,8 @@ public void zip(OutputStream os) throws IOException, InterruptedException {
         zip(os,(FileFilter)null);
     }
 
-    public void zip(FilePath dst) throws IOException, InterruptedException {
+    public void zip(@Nonnull FilePath dst) throws IOException, InterruptedException {
+        assert dst != null;
```

Comment:
I do not think asserts are really required, static analysis should start producing issues even before the tests run (we also need to configure surefire to do the same if possible, like IDEA does)